test/e2e2/spec/test_dyndns.py

```diff
@@ -41,16 +41,21 @@ def do_test(url, headers, params, expected_ipv4, expected_ipv6, subname):
             }
             for qtype in ['A', 'AAAA']
         }
-        rrs_dns = {
-            qtype: {rr.to_text() for rr in NSLordClient.query(params.get('hostname', domain), qtype) or []}
-            for qtype in ['A', 'AAAA']
-        }
 
         for expected_net, qtype in [(expected_ipv4, 'A'), (expected_ipv6, 'AAAA')]:
             assert len(rrs_api[qtype]) == (1 if expected_net else 0)
-            assert len(rrs_dns[qtype]) == (1 if expected_net else 0)
             assert _ips_in_network(rrs_api[qtype], expected_net)
-            assert _ips_in_network(rrs_dns[qtype], expected_net)
+
+            def assertion(query):
+                rrs_dns = {
+                    qt: {rr.to_text() for rr in query(params.get('hostname', domain), qtype) or []}
+                    for qt in ['A', 'AAAA']
+                }
+                assert len(rrs_dns[qtype]) == (1 if expected_net else 0)
+                assert _ips_in_network(rrs_dns[qtype], expected_net)
+                return True
```

Comment:
no `return` needed

Reply:
That's what I initially thought, too. But: This function is called in conftest.py:484 by `assert assertion(**(assertion_kwargs or dict()))`. No return statement leads to `assert None`, which of course fails.

In most cases, the `assertion` function given to `assert_all_ns` returns a Boolean value (actually, something that is truthy or falsey) and so not assert themselves anything. But here, I used the two legacy assert statements because (a) there were already there and (b) it seemed cleaner than `and`-ing them.

Reply:
I think `assert_eventually` should then have a flag to decide whether to wrap its first argument in `_assert` or not.

Reply:
I've addressed this in a fixup commit